server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -85,13 +86,33 @@ long createSecretSeries(
       @Nullable String type,
       @Nullable Map<String, String> generationOptions,
       long now) {
+    long generatedId = rowHmacGenerator.getNextLongSecure();
+    return createSecretSeries(
+        generatedId,
+        name,
+        ownerId,
+        creator,
+        description,
+        type,
+        generationOptions,
+        now);
+  }
+
+  @VisibleForTesting
+  long createSecretSeries(
+      long id,
+      String name,
+      Long ownerId,
+      String creator,
+      String description,
+      @Nullable String type,
+      @Nullable Map<String, String> generationOptions,
+      long now) {
```

Comment:
This is an override of the existing method (which now delegates to this one) with the only difference being that the ID is passed in rather than generated. In order to write a "known-answer test" for the row HMAC we need total control of the inputs (the ID and name), and as it stood the ID was being generated randomly. This seemed more straightforward than trying to futz with the random number generator to spit out some pre-determined output.